packages/core/botpress/src/util.js

```diff
@@ -123,6 +127,49 @@ const getPackageName = pkg => {
   }
 }
 
+const versionExists = version => {
+  if (version == undefined) {
+    throw "Version doesn't exist in botfile.js"
+  }
+
+  return version
+}
+const isValidString = version => {
+  if (_.isEmpty(version) || !_.isString(version)) {
+    throw 'Version must be non-empty string'
+  }
+
+  return version
+}
+const isValidFormat = version => {
+  try {
+    // TODO: change this method if "semver" module will implement semver.isValid()
+    semver.valid(version)
+  } catch (err) {
+    throw 'Version must have semver format (e.g. 10.25.0)'
+  }
+
+  return version
+}
+const isEqualOrGreater = usedMajorVersion => version => {
+  const botfileMajorVersion = semver.major(version)
```

Comment:
It still does not implement what I've quoted, now the 2nd part is missing

```
Check that the major version of bp is exactly the same as of the version from the botfile.

->> Also check that the version of bp is greater than or equal to the version from the botfile.
```

We need **both** checks:
- major versions are equal
- bp version is >= botfile version

Because under the same major version newer versions of bp will be backward-compatible, but older may not be due to unsupported new features, and with the major version change there are breaking changes

Reply:
But you didn't fix it :(
We need the major versions equality (what you used to have before) AND ALSO **full** versions being `bp.version >= botfile.version` where to check for `>=` you use `semver.gte`

Reply:
https://semver.org/ for reference to undersrand why such requirements (combined with my explanation in the previous comment)

Reply:
@emirotin please check again)